torch/csrc/jit/script/compiler.cpp

```diff
@@ -1499,11 +1553,27 @@ struct to_ir {
 
   std::shared_ptr<SugaredValue> emitApplyExpr(Apply &apply, size_t n_binders) {
     auto sv = emitSugaredExpr(apply.callee(), 1);
-    auto inputs = getNamedValues(apply.inputs(), true);
     auto attributes = fmap(apply.attributes(), [&](const Attribute& attr) {
       return NamedValue(attr.range(), attr.name().name(), emitExpr(attr.value()));
     });
-    return sv->call(apply.callee().range(), method, inputs, attributes, n_binders);
+
+    auto loc = apply.callee().range();
+    if (sv->kind() == "fork") {
+      JIT_ASSERT(attributes.size() == 0);
+      JIT_ASSERT(n_binders == 1);
+      auto& trees = apply.inputs().tree()->trees();
+      if (trees.size() != 2) {
+        throw ErrorReport(loc) << "Expected a module and a single tuple argument to fork()";
+      }
+
+      auto sugared_expr = emitSugaredExpr(Expr(trees[0]), 1);
+      TreeList sliced_trees(trees.begin() + 1, trees.end());
+      auto& input = getNamedValues(sliced_trees, true)[0];
```

Comment:
every one of these functions that works on lists has a single element equivalent that should be called when you only have one thing.